src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -166,9 +182,8 @@ public Object createModule(String name) throws CheckstyleException {
                 instance = createObjectFromMap(name, thirdPartyNameToFullModuleNames);
             }
         }
-
         if (instance == null) {
-            instance = createObject(name);
+            instance = createFromFullName(name);
         }
         if (instance == null) {
```

Comment:
can we make code like:
```
         if (instance == null) {
            instance = createObject(name);
         }
         if (instance == null && bruteForce) {
            instance = createFromFullName(name);
         }
         if (instance == null) { 
              // report problem ....
```
to make it clear that bruteforce is last resort.

Reply:
if we do, then here will be violation caused by CyclomaticComplexityCheck

Reply:
@kazachka ,
CyclomaticComplexityCheck only tells that whole method become a bit complicated, but it does not force you to do mess in code that you introduce.
Each comment over code show you that it first candidate to be moved to method.
I would move to special method last block of problem reporting - "createUnableInstantiateException", ... or any other big IF block in this method.